python/lsst/prodstatus/DRPUtils.py

```diff
@@ -0,0 +1,953 @@
+        """Extract elements of the BPS yaml file needed for other functions.
+
+        Parameters
+        ----------
+        bps_yaml_file : `str`
+            File name for yaml file with BPS connection data.
+        ts: `str`
+            Timestamp in %Y%m%dT%H%M%SZ format, or "0" to use first
+            available time.
+
+        Returns
+        -------
+        bpsstr : `str`
+            Description of the BPS connection data.
+        kwd : `dict` [`str`, `str`]
+            Some values extracted from the yaml file.
+        akwd : `dict`
+            Additional data, mostly extracted from other files
+            pointed to by the provided yaml file.
+        ts : `str`
+            Timestamp in %Y%m%dT%H%M%SZ format
+        """
+        kwlist = ["campaign", "project", "payload", "pipelineYaml", "extraQgraphOptions"]
+        kw = {
+            "payload": [
+                "payloadName",
+                "butlerConfig",
+                "dataQuery",
+                "inCollection",
+                "sw_image",
+                "output",
+            ]
+        }
+        f = open(bps_yaml_file, "r")
+        d = load(f, Loader=FullLoader)
+        f.close()
+        kwd = dict()
+        bpsstr = "BPS Submit Keywords:\n{code}\n"
+        for k, v in d.items():
+            if k in kwlist:
+                if k in kw:
+                    for k1 in kw[k]:
+                        kwd[k1] = v[k1]
+                        bpsstr += str(k1) + ":" + str(v[k1]) + "\n"
+                else:
+                    kwd[k] = v
+                    bpsstr += str(k) + ": " + str(v) + "\n"
+        uniqid = "./" + os.path.dirname(bps_yaml_file) + "/submit/" + kwd["output"]
+        for k in kwd:
+            v = kwd[k]
+            uniqid = uniqid.replace("{" + str(k) + "}", v)
+        print(uniqid)
+        if ts == "0":
+            allpath = glob.glob(uniqid + "/*")
+            allpath.sort()
+            longpath = allpath[-1]
+            ts = os.path.basename(longpath)
+        else:
+            ts = ts.upper()
+            longpath = uniqid + "/" + ts
+        # print(longpath)
+        submittedyaml = kwd["output"] + "_" + ts
+        for k in kwd:
+            v = kwd[k]
+            submittedyaml = submittedyaml.replace("{" + str(k) + "}", v)
+        submittedyaml = submittedyaml.replace("/", "_")
+        fullbpsyaml = longpath + "/" + submittedyaml + "_config.yaml"
+        # print(fullbpsyaml)
+        origyamlfile = longpath + "/" + os.path.basename(bps_yaml_file)
+        bpsstr = bpsstr + "bps_submit_yaml_file: " + str(bps_yaml_file) + "\n"
+        akwd = dict()
+        if os.path.exists(origyamlfile):
+            (
+                mode,
+                ino,
+                dev,
+                nlink,
+                uid,
+                gid,
+                size,
+                atime,
+                origyamlfilemtime,
+                ctime,
+            ) = os.stat(origyamlfile)
```

Comment:
This is used to calculate the amount of time it takes to construct the quantum graph when a complete (full) yaml bps submit file is not available.  Usually the full yaml file is there and can be used to get a last modified time for the yaml file, which is compared with the last modified time of the quantum graph file.  The difference is the amount of time it takes to calculate the quantum graph.  If the full yaml file is not available, this will use instead the last modified time of the short version of the bps submit yaml file (the assumption is that this file is copied to the operator's submit directory shortly before the bps submit is executed.  If the file is pre-made long ago, then this doesn't work).

Reply:
Thanks. Some comments to that effect in the code would be very helpful. It would be clearer if you did the stat and only retained the mtime value rather than confusing things with variables that aren't used anywhere.

@MichelleGower this use case suggests that we should consider having bps write out some statistics into a JSON file when it is done. Trying to work things out from modification time seems fragile.

Reply:
Please make JIRA ticket(s) with what we want bps to do.  bps already prints some information using the logger.  We had talked about trying to get the submit output into the google real-time log.  bps can save information into a json file, but it could also do the JIRA interactions automatically (based upon submit yaml values).  We can have the submit process interact with JIRA and presumably the final (merge) job as well.